test_conformance/images/kernel_read_write/test_cl_ext_image_buffer.hpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef _TEST_CL_EXT_IMAGE_BUFFER
+#define _TEST_CL_EXT_IMAGE_BUFFER
+
+#define TEST_IMAGE_SIZE 20
+
+#define GET_EXTENSION_FUNC(platform, function_name)                            \
+    function_name##_fn function_name = reinterpret_cast<function_name##_fn>(   \
+        clGetExtensionFunctionAddressForPlatform(platform, #function_name));   \
+    if (function_name == nullptr)                                              \
+    {                                                                          \
+        return TEST_FAIL;                                                      \
+    }                                                                          \
+    do                                                                         \
+    {                                                                          \
+    } while (false)
+
+static inline size_t aligned_size(size_t size, size_t alignment)
+{
+    return (size + alignment - 1) & ~(alignment - 1);
+}
+
+static inline void* aligned_ptr(void* ptr, size_t alignment)
+{
+    return (void*)(((uintptr_t)ptr + alignment - 1) & ~(alignment - 1));
+}
+
+static inline size_t get_format_size(cl_context context,
+                                     cl_image_format* format)
+{
+    cl_image_desc image_desc = { 0 };
+    image_desc.image_type = CL_MEM_OBJECT_IMAGE1D;
```

Comment:
The image type here matters. We are seeing an issue where the format is `CL_DEPTH` `CL_UNORM_INT16` which is only required for `CL_MEM_OBJECT_IMAGE2D` or `CL_MEM_OBJECT_IMAGE2D_ARRAY`, but the requested type is `CL_MEM_OBJECT_IMAGE1D`.

Can we pass in the image descriptor, or at least `cl_image_desc->image_type` as an argument to `get_format_size` so that the matching image type can be created?

Reply:
Done